libraries/AP_GPS/AP_GPS_GSOF.cpp

```diff
@@ -373,4 +167,49 @@ AP_GPS_GSOF::validate_com_port(const uint8_t com_port) const {
     }
 }
 
+void
+AP_GPS_GSOF::pack_state_data() {
+    // TODO should we pack time data if there is no fix?
+    state.time_week_ms = pos_time.time_week_ms;
+    state.time_week = pos_time.time_week;
+    state.num_sats = pos_time.num_sats;
+
+    if ((pos_time.pos_flags1 & 1)) { // New position
+        state.status = AP_GPS::GPS_OK_FIX_3D;
+        if ((pos_time.pos_flags2 & 1)) { // Differential position 
+            state.status = AP_GPS::GPS_OK_FIX_3D_DGPS;
+            if (pos_time.pos_flags2 & 2) { // Differential position method
+                if (pos_time.pos_flags2 & 4) {// Differential position method
+                    state.status = AP_GPS::GPS_OK_FIX_3D_RTK_FIXED;
+                } else {
+                    state.status = AP_GPS::GPS_OK_FIX_3D_RTK_FLOAT;
+                }
+            }
+        }
+    } else {
+        state.status = AP_GPS::NO_FIX;
+    }
+
+    state.location.lat = (int32_t)(RAD_TO_DEG_DOUBLE * position.latitude * (double)1e7);
+    state.location.lng = (int32_t)(RAD_TO_DEG_DOUBLE * position.longitude * (double)1e7);
+    state.location.alt = (int32_t)(position.altitude * 100);
+    state.last_gps_time_ms = AP_HAL::millis();
+
+    if ((vel.velocity_flags & 1) == 1)
```

Comment:
nitpick: although here we would normally put the curly brace on the same line as the if.

Reply:
I'm using ArduPilot's Astyle formatter on the code, and encorcing it in CI. Do you know how to get the formatter to fix this?

Reply:
Oh nevermind, this is the GPS, which is not autoformatted.